src/services/refactors/extractSymbol.ts

```diff
@@ -605,6 +610,19 @@ namespace ts.refactor.extractSymbol {
         }
     }
 
+    /**
+     * Includes the final semicolon so that the span covers statements in cases where it would otherwise
+     * only cover the declaration list.
+     */
+    function getAdjustedSpanFromNodes(startNode: Node, endNode: Node, sourceFile: SourceFile): TextSpan {
+        const start = startNode.getStart(sourceFile);
+        let end = endNode.getEnd();
+        if (sourceFile.text.charCodeAt(end) === CharacterCodes.semicolon) {

```

Comment:
Would it be interesting to repeat this check as long as `charCodeAt(end)` is a semicolon?

Reply:
That would be undesirable. This refactoring behaves differently when multiple statements are selected.

`var x = 1;`  gets the initializer extracted.

`var x = 1;;` would be treated the same way as `var x = 1; var y = 2` and would be extracted to a new function.

The extra semicolon is obviously a mistake, and while we do currently do the second extraction today if it is selected, we shouldn't otherwise expand the selection to include it.